hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/OMClientRequest.java

```diff
@@ -127,7 +127,14 @@ public OMRequest getOmRequest() {
     }
 
     if (remoteAddress != null) {
-      userInfo.setHostName(remoteAddress.getHostName());
+      OzoneManagerProtocolProtos.UserInfo userInfoInProtocol = getOmRequest()
+          .toBuilder().getUserInfo();
+      if (userInfoInProtocol.getHostName().isEmpty()) {
```

Comment:
If userInfoInProtocol.getHostName(), we need to set it as before right?
Like userInfo.setHostName(remoteAddress.getHostName());

As when someone uses RpcClient, this if check will not be true. So if HostName is empty from UserInfo, I think we still need to set hostname as before.